x-pack/plugins/infra/server/lib/create_custom_metrics_aggregations.ts

```diff
@@ -17,13 +17,22 @@ export const convertToMetricExplorerCustomMetric = (
     return { ...omit(metric, 'aggType'), aggregation: metric.aggType };
   });
 
+const isMetricExpressionCustomMetric = (
+  subject: any[]
+): subject is MetricExpressionCustomMetric[] => {
+  return subject.every((m) => m.aggType != null);
+};
+
 export const createCustomMetricsAggregations = (
   id: string,
-  customMetrics: MetricsExplorerCustomMetric[],
+  customMetrics: MetricsExplorerCustomMetric[] | MetricExpressionCustomMetric[],
   equation?: string
 ) => {
+  const metrics = isMetricExpressionCustomMetric(customMetrics)
+    ? convertToMetricExplorerCustomMetric(customMetrics)
+    : customMetrics;
   const bucketsPath: { [id: string]: string } = {};
-  const metricAggregations = customMetrics.reduce((acc, metric) => {
+  const metricAggregations = metrics.reduce((acc, metric) => {
     const key = `${id}_${metric.name}`;
     if (metric.aggregation === 'count') {
```

Comment:
Just an idea/suggestion. I was reading it, and thought that the part where `MetricsExplorerCustomMetric` is converted to `MetricExpressionCustomMetric` could be simplified. 

 It seems that the main thing for the conversion, would be the `aggregation` info, which is stored in those types, in different attributes.

```ts
const isMetricExpressionCustomMetric = (
  subject: MetricsExplorerCustomMetric | MetricExpressionCustomMetric
): subject is MetricExpressionCustomMetric => {
  return (subject as MetricExpressionCustomMetric).aggType != null;
};

export const createCustomMetricsAggregations = (
  id: string,
  customMetrics: Array<MetricsExplorerCustomMetric | MetricExpressionCustomMetric>,
  equation?: string
) => {
  const bucketsPath: { [id: string]: string } = {};
  const metricAggregations = customMetrics.reduce((acc, metric) => {
    const key = `${id}_${metric.name}`;
    // Instead of converting to MetricsExplorerCustomMetric, we could just run this check here.
    const aggregation = isMetricExpressionCustomMetric(metric)
      ? metric.aggType
      : metric.aggregation;

    // We could now use `aggregation` instead `metric.aggregation`
    if (aggregation === 'count') {
      ...
```

Reply:
Thanks for the suggestion, I applied it on my end.